src/stream.rs

```diff
@@ -6,17 +6,26 @@ pub struct ClonableStream<T: Read + Write>(Arc<Mutex<T>>);
 
 impl<T: Read + Write> Read for ClonableStream<T> {
     fn read(&mut self, buf: &mut [u8]) -> io::Result<usize> {
-        self.0.lock().unwrap().read(buf)
+        self.0
+            .lock()
+            .map_err(|_| io::Error::from(io::ErrorKind::BrokenPipe))?
```

Comment:
I agree that `BrokenPipe` is the error variant that more closely maps to a `PoisonError`, but I would at least log what's happening here first before returning, to avoid confusing the user.

Same for `write()` and `flush()` below

Reply:
Agree, will update